src/problems/maze.clj

```diff
@@ -0,0 +1,38 @@
+(ns problems.maze)
+
+(defrecord State [board current-pos])
+
+(def x "x") ;Pay no attention to the x behind the curtain. This varibale lets us use x without quotes.
+
+(defn legal? [new-pos board]
+  (let [board-rows (count board)
+        board-cols (count (first board))]
+  ;(println (and (>= (first new-pos) 0) (> (get-in board new-pos) 0)))
```

Comment:
You would typically want to get rid of prints like this before you "turn it in" (or submit a pull request).